### PR TITLE
Add aligned-spin conditions to PhenomPv2 angle calculations

### PR DESCRIPTION
**Pull request by SSL32081**
This PR adds back the special treatments for two angles in the aligned-spin case implemented in `LALSimulation` for `PhenomPv2`. 
These changes do not affect the behaviour of `PhenomPv2` in most cases, and only in aligned-spin cases. 

Specifically, they are here in `LALSimulation`:
* `phiJ_sf`: [this line](https://lscsoft.docs.ligo.org/lalsuite/lalsimulation/_l_a_l_sim_i_m_r_phenom_p_8c_source.html#l00372), and
* `alpha0`: [this line](https://lscsoft.docs.ligo.org/lalsuite/lalsimulation/_l_a_l_sim_i_m_r_phenom_p_8c_source.html#l00410)

This PR also fixes the apparent $\pi / 2$ phase shift mentioned in #35.

### CODE REVIEW

File: src/ripplegw/waveforms/IMRPhenomPv2_utils.py [hunk 1]
```diff
@@ -92,7 +94,12 @@ def convert_spins(
 
     thetaJ_sf = jnp.arccos(J0z_sf / J0)
 
-    phiJ_sf = jnp.arctan2(J0y_sf, J0x_sf)
+    no_inplane_J0 = jnp.abs(J0x_sf) < MAX_TOL_ATAN and jnp.abs(J0y_sf) < MAX_TOL_ATAN
+    phiJ_sf = jax.where(
```
Comment (thomasckng) [inline src/ripplegw/waveforms/IMRPhenomPv2_utils.py]: Should be `jnp.where` instead of `jax.where`

**Review by thomasckng — CHANGES_REQUESTED** [review]
Not sure how unit tests work in this repo. Maybe just paste some test cases and results in this PR?

File: src/ripplegw/waveforms/IMRPhenomPv2_utils.py (hunk 1 above)
Comment (SSL32081) [reply src/ripplegw/waveforms/IMRPhenomPv2_utils.py]: Thanks! Just fixed~

**Comment by SSL32081** [comment]
> Not sure how unit tests work in this repo. Maybe just paste some test cases and results in this PR?

Yeah, I'm thinking of pasting the tests I did here, but I’m not sure where to put them. 
Besides the tests regarding the match of waveforms, what else should I include?
Like:
- Jit?
- Gradient?
